lib/elixir/lib/kernel.ex

```diff
@@ -1570,7 +1570,7 @@ defmodule Kernel do
   """
   @spec inspect(Inspect.t, Keyword.t) :: String.t
   def inspect(arg, opts \\ []) when is_list(opts) do
-    opts  = struct(Inspect.Opts, opts)
+    opts  = struct!(Inspect.Opts, opts)
```

Comment:
No, i think it is fine. Most of the Elixir APIs won't warn if you pass extra options so it does not force you to always prune it.